### PR TITLE
Use built local on CI and not LKG

### PR DESCRIPTION
**Pull request by weswigham**
Should prevent the nightly publish from failing again from us being unable to built ourselves with the current snapshot again~

### CODE REVIEW

**Comment by weswigham** [comment]
Note: This PR should fail until I add another commit to fix the build break.

**Comment by weswigham** [comment]
Okay, and that should fix our build.

@ahejlsberg should we look into why
```ts
    export function assign<T extends object>(t: T, ...args: (T | undefined)[]) {
        for (const arg of args) {
            for (const p in arg!) {
                if (hasProperty(arg!, p)) {
                    t![p] = arg![p]; // TODO: GH#23368
                }
            }
        }
        return t;
    }
```
no longer typechecks (or should we call this acceptable)? I'd assume the cause if one of #29847, #29787, or #29858 - I'd wager the last, given this is a function with a rest parameter. It _seems_ like we're wrapping the type with an excess `NonNullable` call which, since they don't compose and simplify, makes the assignment fail.

File: src/compiler/core.ts [hunk 1]
```diff
@@ -1396,9 +1396,10 @@ namespace ts {
 
     export function assign<T extends object>(t: T, ...args: (T | undefined)[]) {
         for (const arg of args) {
-            for (const p in arg!) {
-                if (hasProperty(arg!, p)) {
-                    t![p] = arg![p]; // TODO: GH#23368
+            if (arg === undefined) continue;
+            for (const p in arg) {
+                if (hasProperty(arg, p)) {
+                    t[p] = arg[p]; // TODO: GH#23368
```
Comment (rbuckton) [inline src/compiler/core.ts]: Do we need the TODO anymore?

Comment (weswigham) [reply src/compiler/core.ts]: mm the comment isn't relevant, but the issue still stands.

**Comment by ahejlsberg** [comment]
@weswigham I suspect the build break was caused by #27697, but a contributing factor appears to be some code that was added in #28348 to add non-nullability to the right hand expression in a for-in statement. That code behaves differently than a corresponding narrowing comparison with `null` or `undefined` which is why you could fix the issue by adding an `if` statement.

**Comment by weswigham** [comment]
Mmm very possibly, we don't publish nightlies on weekends, but I forgot to include the things merged over the weekend in the candidates for the failure. In any case, I believe https://github.com/Microsoft/TypeScript/pull/29437 plus the enhancements to empty intersection calculations with type parameters in the negated types PR would fix the issue (as `NonNullable<T>` would simplify to `T` as `T extends object`).

**Pull request merged**